tests/test_hawkdovevar.py

```diff
@@ -0,0 +1,150 @@
+import statistics
+from unittest.mock import patch, Mock
+
+import pytest
+
+from simulatingrisk.hawkdovevar.model import (
+    HawkDoveVariableRiskModel,
+    HawkDoveVariableRiskAgent,
+)
+
+
+def test_init():
+    model = HawkDoveVariableRiskModel(5)
+    # defaults
+    assert model.risk_adjustment is None
+    assert model.hawk_odds == 0.5
+    assert model.include_diagonals is True
+    # unused but should be set to default
+    assert model.adjust_round_n == 10
+
+    # init with risk adjustment
+    model = HawkDoveVariableRiskModel(
+        5,
+        include_diagonals=False,
+        hawk_odds=0.2,
+        risk_adjustment="adopt",
+        adjust_every=5,
+    )
+
+    assert model.risk_adjustment == "adopt"
+    assert model.adjust_round_n == 5
+    assert model.hawk_odds == 0.2
+    assert model.include_diagonals is False
+
+    # handle string none for solara app parameters
+    model = HawkDoveVariableRiskModel(5, risk_adjustment="none")
+    assert model.risk_adjustment is None
+
+    # complain about invalid adjustment type
+    with pytest.raises(ValueError, match="Unsupported risk adjustment 'bogus'"):
+        HawkDoveVariableRiskModel(3, risk_adjustment="bogus")
```

Comment:
These easy access object property-methods really make using and testing quite easy and legible.